yjit/src/options.rs

```diff
@@ -300,7 +326,15 @@ pub fn parse_option(str_ptr: *const std::os::raw::c_char) -> Option<()> {
             }
 
             match opt_val {
-                "" => unsafe { OPTIONS.dump_disasm = Some(DumpDisasm::Stdout) },
+                "" => {
+                    unsafe { OPTIONS.dump_disasm = Some(DumpDisasm::Stdout) };
+                    extern "C" { fn isatty(fd: c_int) -> c_int; }
+                    let stdout = 1;
+                    let is_terminal = unsafe { isatty(stdout) } != 0;
+                    if is_terminal {
```

Comment:
Here I would suggest wrapping this logic into a function in `yjit/src/utils.rs`. Maybe `stdout_is_tty() -> bool`.

I also think that the rigorous way to do this would be to also check if the terminal supports color, but this is probably more rigorous than what we are currently doing (not checking at all).